src/TACSAssembler.cpp

```diff
@@ -4334,11 +4336,13 @@ void TACSAssembler::assembleJacobian(TacsScalar alpha, TacsScalar beta,
 
       // Add the contribution to the residual and the Jacobian from the
       // auxiliary elements - if any, this is scaled by the loadFactor lambda
-      while (aux_count < naux && aux[aux_count].num == i) {
-        aux[aux_count].elem->addJacobian(i, time, alpha * lambda, beta * lambda,
-                                         gamma * lambda, elemXpts, vars, dvars,
-                                         ddvars, elemRes, elemMat);
-        aux_count++;
+      if (aux_count < naux) {
```

Comment:
Are these statements strictly necessary? I thought C++ uses lazy evaluation of logic statements, so if `aux_count < naux` is false then `aux[aux_count].num == i` won't be evaluated and thus there's no risk of indexing past the bounds of `aux`

Reply:
I've run into segfaults caused by this in the past, see this PR conversation [here](https://github.com/smdogroup/tacs/pull/36#discussion_r723684474). I'm not sure if this is a behavior that has changed in newer C++ versions

Reply:
@A-CGray , you seem to be right about short-circuit evaluation in C++. I added those guards more to be consistent with the other locations in the code where the logic was already being used. The code seems to run w/o issues either way. I'm fine  removing them again if needed.

Reply:
OK, up to you, both ways should work